im.c

```diff
@@ -477,6 +480,12 @@ void im_setup(struct wlchewing_state *state) {
 
 	state->chewing = chewing_new();
 	state->xkb_context = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
+	const struct xkb_rule_names default_names = {0};
+	struct xkb_keymap *keymap = xkb_keymap_new_from_names(
+		state->xkb_context, &default_names,
```

Comment:
In https://xkbcommon.org/doc/current/structxkb__rule__names.html:
> If the entire struct is NULL, then each field is taken to be NULL.

In https://github.com/xkbcommon/libxkbcommon/blob/master/NEWS:
> libxkbcommon 0.3.0 - 2013-04-01
>
> The function xkb_keymap_new_from_names() now accepts a NULL value for
> the 'names' parameter, instead of failing. This is equivalent to passing
> a 'struct xkb_rule_names' with all fields set to NULL.

Therefore, it's more simple to write `xkb_keymap_new_from_names(state->xkb_context, NULL, XKB_KEYMAP_COMPILE_NO_FLAGS);`.